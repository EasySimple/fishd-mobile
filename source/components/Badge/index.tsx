import Badge from './Badge';

export { BadgeProps } from './Badge';

export default Badge;
