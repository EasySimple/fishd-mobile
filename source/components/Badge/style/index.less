@import (reference) '../../../assets/css/index.less';

@badgePrefixClass: fm-badge;

.@{badgePrefixClass} {
  position: relative;
  display: inline-block;
  line-height: @line-height-base;
  vertical-align: middle;

  &-text {
    position: absolute;
    top: -1 * @v-spacing-sm;
    z-index: 10;
    min-width: @icon-size-xs / 2;
    height: @icon-size-xs;
    padding: 0 @h-spacing-sm;
    font-size: @font-size-caption-sm;
    font-family: 'Helvetica Neue', Helvetica, 'PingFang SC', 'Hiragino Sans GB', 'Microsoft YaHei', '\5FAE\8F6F\96C5\9ED1', SimSun, sans-serif;
    line-height: @icon-size-xs;
    text-align: center;
    white-space: nowrap;
    color: @color-text-base-inverse;
    border-radius: 2 * @v-spacing-sm;
    background-color: @brand-important;
    text-rendering: optimizeLegibility;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    transform: translateX(-45%);
    transform-origin: -10% center;

    a {
      color: @color-text-base-inverse;
    }

    p {
      padding: 0;
      margin: 0;
    }
  }

  &-hot &-text {
    background-color: #f96268;
  }

  &-dot {
    position: absolute;
    top: -2 * @border-width-lg;
    z-index: 10;
    width: 0.5 * @link-button-font-size;
    height: 0.5 * @link-button-font-size;
    border-radius: 100%;
    background: @brand-important;
    transform: translateX(-50%);
    transform-origin: 0 center;

    &-large {
      width: @link-button-font-size;
      height: @link-button-font-size;
    }
  }

  &-not-a-wrapper &-text,
  &-not-a-wrapper &-dot {
    position: relative;
    top: auto;
    display: block;
    transform: translateX(0);
  }

  &-corner {
    position: absolute;
    top: 0.5 * @link-button-font-size;
    right: -2 * @link-button-font-size;
    width: 5 * @link-button-font-size;
    padding: 0.5 * @link-button-font-size;
    font-size: @font-size-subhead;
    text-align: center;
    white-space: nowrap;
    color: @color-text-base-inverse;
    background-color: @brand-important;
    transform: rotate(45deg);

    &-wrapper {
      overflow: hidden;
    }
  }
}
