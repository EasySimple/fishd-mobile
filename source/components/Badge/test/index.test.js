import React from 'react';
import { render, mount } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Badge from '../Badge.tsx';

describe('Badge', () => {
  it('renders dot prop correctly', () => {
    const wrapper = render(
      <Badge dot>
        <span style={{ width: '0.52rem', height: '0.52rem', background: '#ddd', display: 'inline-block' }} />
      </Badge>,
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.fm-badge-dot')).toHaveLength(1);
  });

  it('renders text correctly', () => {
    const wrapper = mount(
      <Badge text="券" />,
    );
    expect(wrapper.find('.fm-badge-text')).toHaveLength(1);
    expect(wrapper.find('.fm-badge-text').text()).toBe('券');
  });
});