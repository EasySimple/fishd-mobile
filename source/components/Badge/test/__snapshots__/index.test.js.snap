// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders dot prop correctly 1`] = `
<span
  class="fm-badge"
>
  <span
    style="width:0.52rem;height:0.52rem;background:#ddd;display:inline-block"
  />
  <sup
    class="fm-badge-dot"
  />
</span>
`;
